### PR TITLE
Cli tutorial

### PR DESCRIPTION
**Pull request by akomiqaia**

### CODE REVIEW

**Comment by akomiqaia** [comment]
[component brand documentation](/documentation/components/branding/) is not directed to correct path and you get 404. It is in part 3 of the tutorial. Do we have a new page for this?

**Comment by akomiqaia** [comment]
Also chapters `initialising-component-javascript` and `user configuration` in part 5 of the tutorial mention about `o-autoinit`. I left them for now but would be good to know if they are still used while initialising our js.

File: apps/website/_tutorials/new-component/create-a-new-component-part-9.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+<pre><code class="o-syntax-highlight--bash">
+[type]: [description]
+
+[body]
+</code></pre>
+
+The `type` can be any of `feat`, `fix`, `docs` or `backstage`.
+
+The prefix is used to calculate the semver release level, and the section of the release notes to place the commit message in.
```
Comment (notlee) [inline apps/website/_tutorials/new-component/create-a-new-component-part-9.md:41]:
```suggestion
The prefix is used to calculate the version number to release – according to [semver](https://semver.org/#summary) – and the section of the release notes to place the commit message in.
```

File: apps/website/_tutorials/new-component/create-a-new-component-part-9.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+<pre><code class="o-syntax-highlight--bash">
+| type      | when to use                         | release level | release note section |
+| --------- | ----------------------------------- | ------------- | -------------------- |
+| feat      | a feature has been added            | `minor`       | Feature              |
+| fix       | a bug has been patched              | `patch`       | Bug fixes            |
+| docs      | a change to documentation           | none          | Documentation        |
+| backstage | any changes that aren't user-facing | none          | none                 |
+</code></pre>
+Indicate a breaking change by placing an `!` between the type name and the colon. And we will use `feat!` type to do initial release.
```
Comment (notlee) [inline apps/website/_tutorials/new-component/create-a-new-component-part-9.md:50]:
```suggestion
Indicate a breaking change by placing an `!` between the type name and the colon. We will use `feat!` to do an initial release.
```

File: apps/website/_tutorials/new-component/create-a-new-component-part-9.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+<pre><code class="o-syntax-highlight--bash">git add --all
+git commit -m 'feat!: create my o-example component'
+git push</code></pre>
+
+After pushing your commits you should be able to open a pull request. If you have never open a PR(pull request) you can read more about it on [github documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request)
```
Comment (notlee) [inline apps/website/_tutorials/new-component/create-a-new-component-part-9.md:56]:
```suggestion
After pushing your commits you should be able to open a pull request. If you have never open a PR (pull request) you can read more about it in [github documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request)
```

File: apps/website/_tutorials/new-component/create-a-new-component-part-9.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+_To be able to opon a PR you will need to be part of [origami collaborators](https://github.com/orgs/Financial-Times/teams/origami-collaborators) team and if you are not a member already ask [#origami-support](https://financialtimes.slack.com/archives/C02FU5ARJ) to help you._
```
Comment (notlee) [inline apps/website/_tutorials/new-component/create-a-new-component-part-9.md:58]:
```suggestion
_To be able to opon a PR you will need to be part of [origami collaborators](https://github.com/orgs/Financial-Times/teams/origami-collaborators) team. If you are not a member already ask [#origami-support](https://financialtimes.slack.com/archives/C02FU5ARJ) to help you._
```

**Review by notlee — COMMENTED** [review]
Wow wow, so good to get this updated! Heads up that there's a lot of work to review here so a lot of comments from proof-reading, but that doesn't mean there's a lot to change, it's excellent 😄

File: apps/website/_tutorials/new-component/create-a-new-component-part-1.md [hunk 5]
```diff
@@ -28,13 +29,41 @@ In this tutorial we'll build an Origami component. Our example component will di
...
+[ripgrep](https://github.com/BurntSushi/ripgrep) is used in the component `watch` command for quickly choosing the files to watch for changes.
+<pre><code class="o-syntax-highlight--bash">brew install rg</code></pre>
+#### Clone the repo
+<pre><code class="o-syntax-highlight--bash">git clone git@snoot.club:Financial-Times/origami.git
+cd origami</code></pre>
+#### Install dependencies
+
+`npm install` This will install all the components into the root `node_modules`, allows all the components to [find one another](https://nodejs.org/api/modules.html#loading-from-node_modules-folders).
+
+
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-1.md:65]: oh I shuold have noticed that.

File: apps/website/_tutorials/new-component/create-a-new-component-part-1.md [hunk 6]
```diff
@@ -43,134 +72,117 @@ If you have any questions please contact the Origami team to help make this tuto
...
+See our <a href='https://origami.ft.com/blog/2021/06/01/newsletter/#the-origami-specification-is-no-more'>blogpost on why the Origami specification is deprecated</a>.
 </aside>
 
-## Build Tools
+## Create-component CLI
 
-Origami components are developed using the [Origami Build Tools](https://github.com/Financial-Times/origami-build-tools) (`obt`) command line interface. To work on our component, install Origami Build Tools (`obt`) on your machine by running this command in your terminal:
-
-<pre><code class="o-syntax-highlight--bash">npm install -g origami-build-tools</code></pre>
+Origami components are developed using the [Origami `create-component`](https://github.com/Financial-Times/origami/tree/main/tools/create-component) interactive command line interface. The CLI tool was developed using [Glugun](https://infinitered.github.io/gluegun/#/) and it will generate all the files needed for your component configuration. The component will be created in Origami monorepo under the [components](https://github.com/Financial-Times/origami/tree/main/components) folder.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-1.md:80]: Sure makes it easier to read as well

File: apps/website/_tutorials/new-component/create-a-new-component-part-1.md [hunk 7]
```diff
@@ -43,134 +72,117 @@ If you have any questions please contact the Origami team to help make this tuto
...
+<pre><code class="o-syntax-highlight--bash">npm run watch -w components/o-example</code></pre>
 
-<pre><code class="o-syntax-highlight--bash">obt dev</code></pre>
+Opening the link output by the develop command, for example `localhost:5000`, lists the built component assets in the browser (<abbr title="Hyper Text Markup Language">HTML</abbr>, <abbr title="JavaScript">JS</abbr>, and <abbr title="Cascading Style Sheets ">CSS</abbr> files) for each brand that component supports.
 
-Opening the link output by the develop command, for example `localhost:8999`, lists the built component assets in the browser (<abbr title="Hyper Text Markup Language">HTML</abbr>, <abbr title="JavaScript">JS</abbr>, and <abbr title="Cascading Style Sheets ">CSS</abbr> files).
+_`-w components/o-example` flag is used to run `watch` command inside o-example workspace. Origami monorepo uses [NPM workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces) and if you want to execute certain command in different workspace you can by providing `-w` flag and workspace name_
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-1.md:169]: seems like I skipped all the `the`s :D

File: apps/website/_tutorials/new-component/create-a-new-component-part-3.md [hunk 8]
```diff
@@ -36,13 +37,15 @@ A project chooses a brand globally, meaning all components included in a project
...
+When building a branded Origami component it will generate component assets (<abbr title="Hyper Text Markup Language">HTML</abbr>, <abbr title="JavaScript">JS</abbr>, and <abbr title="Cascading Style Sheets ">CSS</abbr> files) for each brand in `local` folder inside the component directory.
 
-For example to build the `internal` version of our component run the command `obt dev --brand internal`. You should see in our demo the background colour has changed from a wheat colour to a light slate colour. That's because wheat is not part of the [internal brand colour palette](https://registry.origami.ft.com/components/o-colors@5.2.5/?brand=internal#demo-primary-palette). As we used a colour usecase `oColorsByUsecase('box', 'background')` in [part two](/documentation/tutorials/create-a-new-component-part-2), rather than specify a specific colour, it was updated automatically for the internal brand.
+To switch between brands you can simple go back to development URL (e.g `localhost:5000`) and choose correct brand HTML file. Alternatively, you can manually change url end point from `/core-demo` to `/internal-demo`.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-3.md:46]: My bad. Will keep an eye in future

File: apps/website/_tutorials/new-component/create-a-new-component-part-5.md [hunk 9]
```diff
@@ -157,24 +158,24 @@ handleEvent(event) {
 <figure>
 	<img alt="" src="/assets/images/tutorial-new-component/hello-world-demo-13-js.png" />
 	<figcaption>
-        The `o-example` component now displays how many times the button has been clicked. Here it has been clicked over 100 times.
+        The `o-example` component now displays how many times the button has been clicked. Here it has been clicked over 10 times.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-5.md:161]: what if someone actually tried to click 100 times to see the change? :D

File: apps/website/_tutorials/new-component/create-a-new-component-part-7.md [hunk 10]
```diff
@@ -1,148 +1,169 @@
...
 </figure>
 
-## Origami Manifest
-
-The `origami.json` manifest file also contributes to the documentation of a component. As we have seen in previous parts of this tutorial `origami.json` includes a component name, description, keywords, demos, demo descriptions, required polyfills, support contacts, maintenance status, and more information which is displayed in the [Origami Registry](https://registry.origami.ft.com/components/).
-
-It is important to keep `origami.json` up to date throughout the lifecycle of the component.
-
-## Part Eight: Component Lifecycle
+_Don't worry if you are unfamiliar with Github and pull request labels. Later, when releasing our component, we will discuss other available labels and link to helpful Github documentation. For now its useful to know that visual regression tests can be run, even if you're not sure yet how they work._
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-7.md:158]: Did not realise that. Should have been more attentive about this

File: apps/website/_tutorials/new-component/create-a-new-component-part-1.md [hunk 11]
```diff
@@ -28,13 +29,41 @@ In this tutorial we'll build an Origami component. Our example component will di
...
+<pre><code class="o-syntax-highlight--bash">git clone git@snoot.club:Financial-Times/origami.git
+cd origami</code></pre>
+#### Install dependencies
+
+`npm install` This will install all the components into the root `node_modules`, allows all the components to [find one another](https://nodejs.org/api/modules.html#loading-from-node_modules-folders).
+
+
+In addition this tutorial introduces a number of tools and libraries such as [git](https://git-scm.com/), [Sass](https://sass-lang.com/), [sinon.js](https://sinonjs.org/), [storybook](https://storybook.js.org/) etc. We do not cover these in depth but attempt to include brief descriptions and links to relevant documentation so that you may learn separately about the parts which are new to you.
 
 If you have any questions please contact the Origami team to help make this tutorial better &#x1F603;. You can find the team on Slack in <a href="https://financialtimes.slack.com/messages/{{site.data.contact.slack}}" target="_blank">#{{site.data.contact.slack}}</a>.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-1.md]: [c486cf0](https://github.com/Financial-Times/origami/pull/808/commits/c486cf0954f39345bddab7afea1a0d9c2ec66829)

File: apps/website/_tutorials/new-component/create-a-new-component-part-1.md [hunk 12]
```diff
@@ -217,9 +229,8 @@ The `div` tag in our demo may be any HTML tag provided there is a `data-o-compon
 
 In part one we learnt:
 - The [Origami specification](/specification/v1/) is deprecated, but some sections provide a useful reference to create components compatible with Origami tools and services whilst they are updated.
-- The [Origami Build Tools](https://github.com/Financial-Times/origami-build-tools) command line interface is used to help us develop and test components.
-- Specifically, we learnt about the Origami Build Tools `init` command to generate a component to work from when developing a new component.
-- Origami components use `git` source control and are stored remotely on Github.
+- Specifically, we learnt about the `npm run create-component` command to generate a component to work from when developing a new component.
+- Origami components are generated inside `components` directory and are stored remotely on Github.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-1.md:239]: [d287c41](https://github.com/Financial-Times/origami/pull/808/commits/d287c410ad0f26023a8c4e7c78bd648a4a7a1dd2)

File: apps/website/_tutorials/new-component/create-a-new-component-part-4.md [hunk 13]
```diff
@@ -11,17 +11,18 @@ redirect_from:
...
-6. [Testing](/documentation/tutorials/create-a-new-component-part-6/)
-7. [Documentation](/documentation/tutorials/create-a-new-component-part-7/)
-8. [Component Lifecycle](/documentation/tutorials/create-a-new-component-part-8/)
+6. [Storybook](/documentation/tutorials/create-a-new-component-part-6/)
+7. [Testing](/documentation/tutorials/create-a-new-component-part-7/)
+8. [Documentation](/documentation/tutorials/create-a-new-component-part-8/)
+9. [Component Lifecycle](/documentation/tutorials/create-a-new-component-part-9/)
 
-In part four we will create new demos to showcase the themes we created in [part three](/documentation/tutorials/create-a-new-component-part-3). We will also revisit demo boilerplate, including the purpose of the `pa11y` demo you may have already noticed.
+In part four we will create new demos to showcase the themes we created in [part three](/documentation/tutorials/create-a-new-component-part-3).
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-4.md]: [3ae32ce](https://github.com/Financial-Times/origami/pull/808/commits/3ae32ce5ba606f732f182d0d0d7470c75815f9ae)

File: apps/website/_tutorials/new-component/create-a-new-component-part-5.md [hunk 14]
```diff
@@ -67,9 +68,9 @@ For more details see the [JavaScript initialisation](/specification/v1/component
 
 ## User Configuration
 
-The second `init` argument is `opts`, an `Object` of options for the user to configure the component. So users of [o-autoinit](https://registry.origami.ft.com/components/o-autoinit@2.0.4/readme) or the [Origami Build Service](https://www.ft.com/__origami/service/build/v2/) can also configure components, data attributes may alternatively be used to set component configuration.
+The second `init` argument is `options`, an `Object` of options for the user to configure the component. So users of [o-autoinit](https://registry.origami.ft.com/components/o-autoinit@2.0.4/readme) can also configure components, data attributes may alternatively be used to set component configuration.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-5.md]: [7bd3035](https://github.com/Financial-Times/origami/pull/808/commits/7bd30354614ccc9158cd04cbf539591270fc3d91)

File: apps/website/_tutorials/new-component/create-a-new-component-part-7.md [hunk 15]
```diff
@@ -1,148 +1,169 @@
...
+We can't run Percy yet as we haven't released a version of our component to compare changes against. But later, when we have released our component, you will be able to run [percy.io](https://percy.io/) by adding a `percy` label to Github pull requests. When Percy has run a comment is added to the pull request, and the demo comparisons are ready for review at [percy.io/Financial-Times](https://percy.io/Financial-Times/).
 
 <figure>
-	<img alt="" src="/assets/images/tutorial-new-component/hello-world-demo-17-docs.png" />
+	<img alt="" src="/assets/images/tutorial-new-component/hello-world-demo-21-tests.png" />
 	<figcaption>
-        The SassDoc is shown in the Origami registry. This image shows <a href="https://registry.origami.ft.com/components/o-table@8.1.2/sassdoc?brand=core">o-table SassDocs</a>.
+        This image show a Github pull request where the `percy` label had been added. Percy then ran to visually compare the component demos in the pull request against the last release, and removed the `percy` label when done.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-7.md]: [29b898c](https://github.com/Financial-Times/origami/pull/808/commits/29b898c183b1fddafc4f66e9341942858736503c)

File: apps/website/_tutorials/new-component/create-a-new-component-part-6.md [hunk 16]
```diff
@@ -1,169 +1,180 @@
...
-
-## Sass Tests
+But let's brake this down a bit and compare it to the `demo.mustache` file.
+1. First thing we notice is `ExampleProps` [type declaration](https://www.typescriptlang.org/docs/handbook/basic-types.html) for our [component props](https://reactjs.org/docs/components-and-props.html). This type declaration is used to make sure that our component only gets passed down theme prop and it must be string.
+2. Second thing we notice is that we are using the [JSX syntax](https://reactjs.org/docs/jsx-in-depth.html) to write our component and return it from `Example` function.
+3. Syntax for JSX and Mustache are slightly different. In JSX we use `className` instead of `class` and we are passing theme prop to our JSX using literal string (`${theme}`) syntax.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-6.md]: [3d0873c](https://github.com/Financial-Times/origami/pull/808/commits/3d0873cb39ac274cbc56f9ba9d83ebac45fdbe8a)

File: apps/website/_tutorials/new-component/create-a-new-component-part-6.md [hunk 17]
```diff
@@ -1,169 +1,180 @@
...
 
-To demonstrate how these projects are used to test components we will add a new test to confirm that clicking a button in our component increments the count.
+JSX is heavily linked to React and its ecosystem but it JSX is just a syntax that can exists without React and it could be used by other frameworks. The whole idea behind using JSX syntax is to make our storybook demos easy to write and also enable developers to easily import this templates in their code. Our TSX templates don't use React specific imports in them but we are leveraging [JSX transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) which enables us to use JSX without importing React and therefore making it somewhat framework agnostic.
 
-JavaScript tests are located under the `tests/js` directory. The file `example.test.js` already has boilerplate tests, which use component markup defined in `tests/js/helpers/fixtures.js` to confirm the `init` method works as expected.
+We are writing our compononent in [typeScript](https://www.typescriptlang.org/docs/handbook/basic-types.html) which enables us to make less errors.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-6.md]: [3d0873c](https://github.com/Financial-Times/origami/pull/808/commits/3d0873cb39ac274cbc56f9ba9d83ebac45fdbe8a)

File: apps/website/_tutorials/new-component/create-a-new-component-part-6.md [hunk 18]
```diff
@@ -1,169 +1,180 @@
...
 
-Component JavaScript tests use [mocha](https://mochajs.org/) as a test runner; [sinon](https://sinonjs.org/) for stubs, spies, and mocks; and [proclaim](https://github.com/rowanmanning/proclaim) to make assertions.
+The code in `example.tsx` will be very similar to what we have in `demo.mustache` but with different syntax. The syntax we are using is called [JSX](https://reactjs.org/docs/jsx-in-depth.html).
 
-To demonstrate how these projects are used to test components we will add a new test to confirm that clicking a button in our component increments the count.
+JSX is heavily linked to React and its ecosystem but it JSX is just a syntax that can exists without React and it could be used by other frameworks. The whole idea behind using JSX syntax is to make our storybook demos easy to write and also enable developers to easily import this templates in their code. Our TSX templates don't use React specific imports in them but we are leveraging [JSX transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) which enables us to use JSX without importing React and therefore making it somewhat framework agnostic.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-6.md]: [3d0873c](https://github.com/Financial-Times/origami/pull/808/commits/3d0873cb39ac274cbc56f9ba9d83ebac45fdbe8a)

File: apps/website/_tutorials/new-component/create-a-new-component-part-6.md [hunk 19]
```diff
@@ -1,169 +1,180 @@
...
+6. Storybook
+7. [Testing](/documentation/tutorials/create-a-new-component-part-7/)
+8. [Documentation](/documentation/tutorials/create-a-new-component-part-8/)
+9. [Component Lifecycle](/documentation/tutorials/create-a-new-component-part-9/)
 
-In part six we will add tests to our component. Including tests for Sass, JavaScript, and common accessibility issues.
+In part six we will rewrite `demo.mustache` into tsx template, we will use our component's javascript code to initialise interactivity for storybook demos and implement all the variants of our component.
 
-Run `obt test` to run component tests. Run `obt verify` to lint code style and check the validity of `origami.json`.
+## Storybook
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply apps/website/_tutorials/new-component/create-a-new-component-part-6.md:27]: [3d0873c](https://github.com/Financial-Times/origami/pull/808/commits/3d0873cb39ac274cbc56f9ba9d83ebac45fdbe8a)

**Review by notlee — APPROVED** [review]

**Pull request merged**